server/utils/ResultTypes.d.ts

```diff
@@ -1,22 +1,22 @@
 export interface Result<S, E> {
-  value(): S
-  error(): E
+  success(): S
+  fail(): E
   map<T>(f: (v: S) => T): Result<T, E>
-  orElse(f: () => Result<S, E>): Result<S, E>
+  orElseTryAsync(f: () => Promise<this>): Promise<this>
```

Comment:
renamed to `orElseTryAsync`, `orElse` would generally have the type signature: `orElse(f: () => Result<S, E>): S`.

Using polymorphic `this` to return the current type